test/e2e/app-dir/dynamic-io/dynamic-io.test.ts

```diff
@@ -330,4 +331,41 @@ describe('dynamic-io', () => {
       expect($('#page-children').text()).toBe('at buildtime')
     }
   })
+
+  it('should resume when client components are dynamic but the RSC render was static', async () => {
```

Comment:
```suggestion
  it('should not resume when client components are dynamic but the RSC render was static', async () => {
```
(Unless you mean resume in the browser!?)